### DISCUSSION TITLE
IB gateway settings are not persistent

### DISCUSSION THREAD

**Discussion by lkaino** [gnzsnz/ib-gateway-docker#136] [Q&A] [answered]
I find that for example the "Bypass..." settings that I check using a VNC connection through the menus are not persisting. Am I missing some volume mount to the Docker?

**Comment by gnzsnz** [comment]
i can't help you with the level of detail that you provide. I have test it and i can confirm that it works, if it's not the case please provide reproduction steps.

**Reply by lkaino** [reply]
Do you know where the settings are stored? I imagine I need to volume mount that location from the host to the container in order for the settings to persist.

Here are the volumes I'm mounting:
```
volumes:
      - /volume1/docker/ibgateway/ibc/config.ini:/home/ibgateway/ibc/config.ini
      - /volume1/docker/ibgateway/ibc/jts.ini:/home/ibgateway/Jts/jts.ini
      - /volume1/docker/ibgateway/ibc/external_ibc:/mnt/external_ibc
```

**Reply by lkaino** [reply]
![Screenshot_20240725_125751_RVNC Viewer](https://github.com/user-attachments/assets/d937314e-de9c-4aa0-b4a3-2aa19b768ced)

These are the settings I was referring to, just as an example of settings that the app keeps forgetting.

**Reply by gnzsnz** [reply, marked as answer]
Both are covered on the `README`, for bypass there is an environment variable `BYPASS_WARNING` 

To persist settings there is a section covering what is needed in terms of environment variable and volume. Have you followed it?
